os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -1157,54 +1157,93 @@ static void amebasmart_spi_exchange(FAR struct spi_dev_s *dev,
  ************************************************************************************/
 
 	int mode_16bit = 1;
+	uint32_t len = 0;
 	spi_hookmastercallback(priv);
 	if (amebasmart_spi_9to16bitmode(priv)) {
 		/* 16-bit mode */
 		mode_16bit++;
 	}
-	DEBUGASSERT((nwords * mode_16bit) <= SPI_DMA_MAX_BUFFER_SIZE);
-	if (txbuffer && rxbuffer) {
-		uint8_t *rxbuff_aligned = NULL;
+	len = nwords * mode_16bit;
+	uint8_t *rxbuff_aligned = NULL;
+	uint8_t *txbuff_aligned =  NULL;
+	if (len >= SPI_DMA_MAX_BUFFER_SIZE) {
```

Comment:
I understand that max DMA page size is 64KB. However, I think 64KB is too big. Also, as exchange return type is NULL, currently there is no way to know if the transfer has failed because of no memory. Hence, I think it would be good to allocate the tx and rx buffers with spi initialize. Since 64KB * 4 (2 spi ports, 2 buffers tx, rx) = 256KB is a lot, may be we can use a smaller buffers of 4KB (16KB total usage). I think this will also increase the performance a bit as we skip the malloc calls in exchange function.

Reply:
Hi @abhishek-samsung,
So where do we free this malloc, i don't see there is a deinit in spi operation.
Also, how much will the performance increase with skipping the malloc calls in exchange function? I think it will add to the complicity during exchange, as we need to consider which set of tx/rx buffer to use... 
I agree that using 64KB might take up alot of memory, but we also need to consider the fact there will be a slight gap between each of the slicing data (~130us), if we slice it to 4K there will be more gaps, so im afraid this might also affect performance. 
Please share your thoughts, thank you.

Reply:
Okay,
1. I think we should add some logs so that we get to know when SPI transfer failed due to malloc.
2. Okay, lets use 64KB then.

Reply:
Hi @abhishek-samsung ,
I have added lldbg("rxbuff_aligned malloc failed\n") when malloc failed.